File: backend/src/api/article.js
import service from "@/utils/request";

export const createArticle = (data) => {
  return service({
    url: "/article/createArticle",
    method: "post",
    data,
  });
};

export const deleteArticle = (params) => {
  return service({
    url: `/article/deleteArticle/${params.ID}`,
    method: "delete",
  });
};

export const deleteArticleByIds = (data) => {
  return service({
    url: "/article/deleteArticleByIds",
    method: "delete",
    data,
  });
};

export const updateArticle = (data) => {
  return service({
    url: `/article/updateArticle/${data.ID}`,
    method: "put",
    data,
  });
};

export const findArticle = (params) => {
  return service({
    url: `/article/findArticle/${params.ID}`,
    method: "get",
  });
};

export const getArticleList = (params) => {
  return service({
    url: "/article/getArticleList",
    method: "get",
    params,
  });
};
// 上传文件

export const uploadFile = (params) => {
  return service({
    url: "/fileUploadAndDownload/upload",
    method: "post",
    data: params.form,
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
};

//批量更新 是否首页显示
export const putArticleByIds = (data) => {
  return service({
    url: "/article/putArticleByIds",
    method: "put",
    data: data,
  });
};

// 文章阅读量
export const getReading = () => {
  return service({
    url: "/article/getArticleReading",
    method: "get",
  });
};

File: server-fiber/router/app/article.go
package app

import (
	v1 "server-fiber/api/v1"
	"server-fiber/middleware"

	"github.com/gofiber/fiber/v2"
)

type ArticltRouter struct{}

// InitArticleRouter 初始化 article 路由信息
func (s *ArticltRouter) InitArticleRouter(Router *fiber.App) {
	articleRouter := Router.Use(middleware.OperationRecord())

	var articleApi = v1.ApiGroupApp.AppApiGroup.ArticleApi
	{
		articleRouter.Post("createArticle", articleApi.CreateArticle)             // 新建article
		articleRouter.Delete("DeleteArticle/:id", articleApi.DeleteArticle)       // 删除article
		articleRouter.Delete("DeleteArticleByIds", articleApi.DeleteArticleByIds) // 批量删除article
		articleRouter.Put("updateArticle/:id", articleApi.UpdateArticle)          // 更新article
		articleRouter.Put("PutArticleByIds", articleApi.PutArticleByIds)          // 批量更新 是否首页显示article
	}
	{
		Router.Get("findArticle/:id", articleApi.FindArticle)         // 根据ID获取article
		Router.Get("GetArticleList", articleApi.GetArticleList)       // 获取article列表
		Router.Get("GetArticleReading", articleApi.GetArticleReading) // 获取阅读量
	}
}
